### ISSUE TITLE
npm 5 and symlink issues - Ideas how to solve?

### ISSUE THREAD

**Issue by Bubelbub** [npm/npm#17884]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [X] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

There is an NPM package which is called `cordova-plugin-device` and each time I try to install it I get this error.

```
+ npm install
npm WARN checkPermissions Missing write access to /Users/vagrant/git/node_modules/cordova-plugin-device
netz-app@ /Users/vagrant/git
└── cordova-plugin-device@1.1.7-dev  (git+https://github.com/DDTChen/cordova-plugin-device.git#b0965e2a5cc7fca2748a7d67e1b311962730043e)
npm WARN ionic-gallery-modal@0.1.1 requires a peer of ionic-angular@3.3.0 but none was installed.
npm WARN ionic-img-viewer@2.4.0 requires a peer of ionic-angular@~3.4.0 but none was installed.
npm WARN ngx-chips@1.4.6 requires a peer of @angular/common@^4.2.0 but none was installed.
npm WARN ngx-chips@1.4.6 requires a peer of @angular/core@^4.2.0 but none was installed.
npm WARN ngx-chips@1.4.6 requires a peer of @angular/forms@^4.2.0 but none was installed.
npm WARN ngx-chips@1.4.6 requires a peer of @angular/animations@^4.2.0 but none was installed.
npm WARN ngx-tooltip@0.0.9 requires a peer of @angular/core@^2.0.0 but none was installed.
npm ERR! path /Users/vagrant/git/node_modules/cordova-plugin-device
npm ERR! code ELOOP
npm ERR! errno -62
npm ERR! syscall access
npm ERR! ELOOP: too many symbolic links encountered, access '/Users/vagrant/git/node_modules/cordova-plugin-device'
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/vagrant/.npm/_logs/2017-07-22T00_37_06_939Z-debug.log
```

So I'm open this issue because I want to know what could be the reason for this `ELOOP: too many symbolic links encountered, access` problem.
The same package is working well on `NPM@4` but not with `NPM@5`.

I searched for Symlinks in this package and I can't find any.
This package is really small and I think normally there can't be a problem with it.
But then I must search the problem on `NPM` side and this is complex so I am afraid to start searching here.

Can someone give me an idea for what I must search or a blog post of how to find issues like this and fix them?
Then I could start searching and fixing it 👍 

#### How can the CLI team reproduce the problem?

- `$ npm -g install npm@5`
- `$ npm install cordova-plugin-device`

### supporting information:

 - `npm -v` prints: 5.3.0
 - `node -v` prints: v6.9.1
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux
 - Network issues:
   - Geographic location where npm was run: Germany, Frankfurt
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by dpogue** [comment]
If you're still able to reproduce this, I've been running into what might be the same bug. [I opened a bug report against Cordova](https://issues.apache.org/jira/browse/CB-13252).

When you run into this error, does your package.json get updated such that `cordova-plugin-device` is pointed to `file:node_modules/cordova-plugin-device`?

**Comment by krlozadan** [comment]
Yes, @dpogue I have the exact same error I' tired of finding a solution. I've lost two complete days trying to get my ionic projet working again I ran out of ideas here.
Everything goes well all the way up to adding any plugin actually.

Here's my ionic info output

cli packages: 
    @ionic/cli-utils  : 1.10.2
    ionic (Ionic CLI) : 3.10.3
global packages:
    Cordova CLI : 7.0.1
local packages:
    @ionic/app-scripts : 2.1.4
    Cordova Platforms  : android 6.2.3
    Ionic Framework    : ionic-angular 3.2.1
System:
    Node  : v6.11.3
    npm   : 5.3.0
    OS    : macOS Sierra
    Xcode : Xcode 9.0 Build version 9A235

**Comment by krlozadan** [comment]
I got it working!!
1.- Removed the node_modules folder and the package-lock.json
2.- Changed manually the package.json file cordova-plugin-device version to ^1.1.6
3.- Ran this command `sudo ionic cordova plugin add cordova-plugin-device --force`

Hope it helps!

**Comment by stevengill** [comment]
so it looks like if I cd into `node_modules` and run `npm install cordova-plugin-device --no-save`, it works as expected (by adding the module under in the `node_modules` folder). Running it a second time will create a symlink in the `node_modules` directory. If I run it from outside of the node_modules directory, no symlink issue
